packages/integrations/tailwind/src/index.ts

```diff
@@ -14,13 +16,59 @@ function getDefaultTailwindConfig(srcUrl: URL) {
 	};
 }
 
-export default function (): AstroIntegration {
+async function getUserConfig(projectRoot: URL, configPath?: string) {
+	const resolvedProjectRoot = fileURLToPath(projectRoot);
+	let userConfigPath: string | undefined;
+
+	if (configPath) {
+		const configPathWithLeadingSlash = /^\.*\//.test(configPath) ? configPath : `./${configPath}`;
+		userConfigPath = fileURLToPath(new URL(configPathWithLeadingSlash, `file://${resolvedProjectRoot}/`));
+	}
+
+	return await load('tailwind', { mustExist: false, cwd: resolvedProjectRoot, filePath: userConfigPath });
+}
+
+type IntegrationConfig =
+	| {
+			config?: {
+				/**
+				 * Path to your tailwind config file
+				 * @default 'tailwind.config.js'
+				 */
+				path?: string;
+				/**
+				 * Apply Astro's default Tailwind config as a preset
+				 * This is recommended to enable Tailwind across all components and Astro files
+				 * @default true
+				 */
+				applyAstroPreset?: boolean;
+			};
+	  }
+	| undefined;
+
+export default function (integrationConfig: IntegrationConfig): AstroIntegration {
+	const applyAstroConfigPreset = integrationConfig?.config?.applyAstroPreset ?? true;
 	return {
 		name: '@astrojs/tailwind',
 		hooks: {
-			'astro:config:setup': ({ config, injectScript }) => {
+			'astro:config:setup': async ({ config, injectScript }) => {
 				// Inject the Tailwind postcss plugin
-				config.styleOptions.postcss.plugins.push(tailwindPlugin(getDefaultTailwindConfig(config.src)));
+				const userConfig = await getUserConfig(config.projectRoot, integrationConfig?.config?.path);
+
+				let tailwindConfig: TailwindConfig;
+				if (typeof userConfig?.value === 'object' && userConfig.value !== null) {
```

Comment:
If `userConfig?.value` returned something but that something wasn't an object, I almost would want this to be an error that the user sees, instead of silently failing this check and using the default config below. 

Can we do the simpler `if (userConfig && userConfig.value) {` here?

Reply:
Actually, taking a step further back: what happens if `options.path` is given to the integration, but then fails to load? I'd expect that to be an error for the user to investigate & fix.


Another good reason to do this is that it should really let you simplify the code below where you're trying to handle this tricky case. For example, I think it would mean that you could just do this, below:

```js
/* throw if options.path was given, but userConfig or userConfig.value failed to load */

const tailwindConfig = userConfing.value ?? getDefaultTailwindConfig(config.src);
if (applyAstroConfigPreset) {
  tailwindConfig.presets.push(getDefaultTailwindConfig(config.src));
}
```